iris/src/iris/pipeline/abstract_agent_pipeline.py

```
@@ -288,6 +289,48 @@ def _run_agent_iterations(
+            )
+            return None
+
+        try:
+            if output:
+                session_title = self.session_title_pipeline(first_user_msg, output)
+                if self.session_title_pipeline.tokens is not None:
+                    self._track_tokens(state, self.session_title_pipeline.tokens)
+                if session_title is None:
+                    logger.error("Generating session title failed.")
+                return session_title
+            return None
```

Ruff (0.14.0)

Consider moving this statement to an `else` block

(TRY300)